build-system/tasks/check-types.js

```diff
@@ -160,6 +158,17 @@ const TYPE_CHECK_TARGETS = {
   },
 };
 
+/**
+ * Object of targets to check with TypeScript.
+ *
+ * @type {Object<string, {tsconfig: string}>}
+ */
+const TSC_TYPECHECK_TARGETS = {
+  'compiler': {
+    tsconfig: 'src/compiler/tsconfig.json',
```

Comment:
generate food for thought: if these targets instead all specify a root directory (assumed to contain a `tsconfig.json` file), then the closure targets could automatically exclude those directories rather than the explicit exclusion (`'!src/compiler/**/*.js`) above. Might reduce the number of  places we have to touch this file in order to add a new target/directory. I don't feel strongly about this.

Reply:
Done